lib/send_with_us/api.rb

```diff
@@ -23,41 +23,103 @@ def initialize(options = {})
       @configuration = SendWithUs::Config.new(settings)
     end
 
-    def send_with(email_id, to, data = {}, from = {}, cc = {}, bcc = {}, files = [], esp_account = '', version_name = '', headers = {}, tags = [])
+    # DEPRECATED: Please use 'send' instead.
+    #
+    # Sends the specified email with any arguments.
+    #
+    # * *Args*    :
+    #   - +email_id+ -> ID of the email to send
+    #   - +to+ -> Hash of recipient details
+    #   - +data+ -> Hash of email data
+    #   - +from+ -> Hash of sender details
+    #   - +cc+ -> Array of CC recipients
+    #   - +bcc+ -> Array of BCC recipients
+    #   - +files+ -> Array of attachments
+    #   - +esp_account+ -> The ESP account to use
+    #   - +version_name+ -> The specific email version to use
+    #   - +headers+ -> Hash of headers
+    #   - +tags+ -> Array of tags
+    #
+    def send_with(email_id, to, data = {}, from = {}, cc = [], bcc = [], files = [], esp_account = '', version_name = '', headers = {}, tags = [])
+      warn "[DEPRECATED] 'send_with' is deprecated.  Please use 'send' instead."
+
+      send(
+        email_id,
+        to,
+        data: data,
+        from: from,
+        cc: cc,
+        bcc: bcc,
+        files: files,
+        esp_account: esp_account,
+        version_name: version_name,
+        headers: headers,
+        tags: tags
+      )
+    end
 
+    # Sends the specified email with any optional arguments
+    #
+    # * *Args*    :
+    #   - +email_id+ -> ID of the email to send
+    #   - +to+ -> Hash of recipient details
+    # * *Options*    :
+    #   - +:data+ -> Hash of email data
+    #   - +:from+ -> Hash of sender details
+    #   - +:cc+ -> Array of CC recipients
+    #   - +:bcc+ -> Array of BCC recipients
+    #   - +:files+ -> Array of attachments
+    #   - +:esp_account+ -> The ESP account to use
+    #   - +:version_name+ -> The specific email version to use
+    #   - +:headers+ -> Hash of headers
+    #   - +:tags+ -> Array of tags
+    #   - +:locale+ -> Localization string
+    #
+    def send(email_id, to, options = {})
```

Comment:
`email_id` and `to` are required fields, which is why they're left out of `options`.

Encapsulating the optional arguments in a class is a possible solution - Ruby's convention of using `options` for optional named arguments was much simpler though.